Tests/unit/core/test_time.cpp

```diff
@@ -21,13 +21,55 @@
 
 #include <gtest/gtest.h>
 #include <core/core.h>
+#include <ctime>
 
 using namespace WPEFramework;
 using namespace WPEFramework::Core;
 static constexpr uint32_t MilliSecondsPerSecond = 1000;
 static constexpr uint32_t MicroSecondsPerMilliSecond = 1000;
 static constexpr uint32_t MicroSecondsPerSecond = MilliSecondsPerSecond * MicroSecondsPerMilliSecond;
 
+
+int32_t GetMachineTimeDifference(string zone)
+{
+    time_t defineTime = 1325376000;
+    setenv("TZ", zone.c_str(), 1);
+    tzset();
+    struct tm *time = localtime(&defineTime);
+    return time->tm_gmtoff;
+}
+
+// NOTE: time TCs executed in UTC time zone and PDT environments
+string GetTimeBasedOnTimeZone(string zone, struct tm time, bool localTime = true)
+{
+    setenv("TZ", zone.c_str(), 1);
+    tzset();
+    time_t rawTime;
+    struct tm convertedTime;
+    char strTime[25];
+    rawTime = mktime(&time);
+    if (localTime) {
+        localtime_r(&rawTime, &convertedTime);
+    } else {
+        gmtime_r(&rawTime, &convertedTime);
+    }
+
+    uint32_t timeRead = convertedTime.tm_hour * 3600 + convertedTime.tm_min * 60 + convertedTime.tm_sec;
+    // NOTE: Currently new Time is calcualting only gmtoff with higher values, since for negative gmtoff case
+    // thunder giving same value, with out subtracting the time difference
+    uint32_t newTime = timeRead + ((convertedTime.tm_gmtoff > 0) ? convertedTime.tm_gmtoff: 0);
```

Comment:
As far as I understand the documentation you should not use this field to actual compensate the struct tm  that was retreived, it is for informational purposes only (it is even a nonstandard field of struct tm) unless you are trying to return UTC time again but I do not think that is the case. Let's have a call tomorrow to discuss because I assume if you do not add to the result your return value differs from that from time.h, correct? It would be good see what the correct value should be in the testcases (e.g. take you local timezone and see of the result of this and thunder is correct or not. But let's call to discuss this.

Reply:
yes updated TCs. Found it is working properly if I use Time(with isLocaltime as false) case getting proper values. So updated TCs